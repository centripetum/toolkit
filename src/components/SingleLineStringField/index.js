import * as React from 'react'
import { not } from 'sanctuary'

import useFieldState from '../../hooks/useFieldState'
import Button from '../styled/Button'
import Input from '../styled/Input'
import Label from '../styled/Label'
import isEmpty from '../../utils/isEmpty'

export default function Field (props) {
  const {
    defaultValue,
    handleChange,
    label,
    name,
    noClear,
    noLabel,
    noReset,
    noUndoRedo,
    onSave
  } = props

  const {
    clear,
    dirty,
    noRedo,
    noUndo,
    onChange,
    redo,
    reset,
    undo,
    value
  } = useFieldState({
    defaultValue,
    handleChange,
    noClear,
    noReset,
    noUndoRedo
  })
  const handleSave = onSave ? () => onSave(name, value) : null

  return (
    <div>
      {noLabel && <Label htmlFor={name}>{label}</Label>}
      <Input name={name} type='text' value={value} onChange={onChange} />
      {undo && (
        <Button onClick={undo} disabled={noUndo}>
          Undo
        </Button>
      )}
      {redo && (
        <Button onClick={redo} disabled={noRedo}>
          Redo
        </Button>
      )}
      {clear && (
        <Button onClick={clear} disabled={isEmpty(value)}>
          Clear
        </Button>
      )}
      {handleSave && (
        <Button onClick={handleSave} disabled={value === defaultValue}>
          Save
        </Button>
      )}
      {reset && (
        <Button onClick={reset} disabled={not(dirty)}>
          Reset
        </Button>
      )}
    </div>
  )
}
