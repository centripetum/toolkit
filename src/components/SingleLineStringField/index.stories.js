import * as React from 'react'
import { storiesOf } from '@storybook/react'

import SingleLineStringField from '.'

storiesOf('SingleLineStringField', module).add('base', () => (
  <SingleLineStringField />
))
