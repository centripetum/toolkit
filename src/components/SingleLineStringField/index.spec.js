import * as React from 'react'
import { render } from 'react-testing-library'

import SingleLineStringField from './'

describe('components:SingleLineStringField', () => {
  it('works', () => {
    const { container } = render(
      <SingleLineStringField
        defaultValue='Everything and log'
        label='Message'
        name='message'
        onSave={() => null}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
