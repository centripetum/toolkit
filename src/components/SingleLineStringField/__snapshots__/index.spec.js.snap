// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:SingleLineStringField works 1`] = `
<div>
  <div>
    <input
      class="sc-bwzfXH lfkgtm"
      name="message"
      type="text"
      value="Everything and log"
    />
    <button
      class="sc-bdVaJa hfjiEQ"
      disabled=""
    >
      Undo
    </button>
    <button
      class="sc-bdVaJa hfjiEQ"
      disabled=""
    >
      Redo
    </button>
    <button
      class="sc-bdVaJa kYzAVN"
    >
      Clear
    </button>
    <button
      class="sc-bdVaJa hfjiEQ"
      disabled=""
    >
      Save
    </button>
    <button
      class="sc-bdVaJa hfjiEQ"
      disabled=""
    >
      Reset
    </button>
  </div>
</div>
`;
